### PR TITLE
Simplify the dummy dataloader

### PR DESCRIPTION
**Pull request by trajepl**
## Checklist before requesting a review
- [x] Add unit tests for this change.
- [x] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [x] Format your code by running `pre-commit run --all-files`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

### CODE REVIEW

File: olive/evaluator/olive_evaluator.py [hunk 1]
```diff
@@ -33,6 +34,8 @@
 
 logger = logging.getLogger(__name__)
 
+# TODO: remove
+
 
 class DummyDataloader(Dataset):
```
Comment (guotuofeng) [inline olive/evaluator/olive_evaluator.py]: >DummyDataloader [](http://example.com/codeflow?start=6&length=15)

do we still need it?

Comment (trajepl) [reply olive/evaluator/olive_evaluator.py]: Already removed since the latest commit. Please refresh it.

File: olive/evaluator/olive_evaluator.py [hunk 2]
```diff
@@ -183,8 +156,14 @@ def get_user_config(metric: Metric):
         eval_func = user_module.load_object(evaluate_func)
 
         if metric.user_config.input_names and metric.user_config.input_shapes and not dataloader and not eval_func:
-            dataloader = DummyDataloader(
-                metric.user_config.input_names, metric.user_config.input_shapes, metric.user_config.input_types
+            dataloader = (
```
Comment (guotuofeng) [inline olive/evaluator/olive_evaluator.py:159]: >( [](http://example.com/codeflow?start=25&length=1)

why do we need the outer side parenthesis?

File: olive/model.py [hunk 3]
```diff
@@ -648,22 +647,24 @@ def get_dummy_inputs(self):
...
-            }
-            input_types = self.io_config.input_types or ["float32"] * len(self.io_config.input_shapes)
-            dummy_inputs = []
-            for shape, dtype in zip(self.io_config.input_shapes, input_types):
-                dummy_inputs.append(torch.zeros(shape, dtype=str_to_type[dtype]))
-            dummy_inputs = tuple(dummy_inputs) if len(dummy_inputs) > 1 else dummy_inputs[0]
-
-        self.dummy_inputs = dummy_inputs
+        elif self.io_config and self.io_config.input_shapes:
+            dummy_inputs, _ = (
```
Comment (guotuofeng) [inline olive/model.py:651]: >( [](http://example.com/codeflow?start=30&length=1)

ditto

File: olive/evaluator/olive_evaluator.py (hunk 2 above)
Comment (trajepl) [reply olive/evaluator/olive_evaluator.py:159]: image

Actually, it was formatted by linter.

File: olive/model.py (hunk 3 above)
Comment (trajepl) [reply olive/model.py:651]: Same.

**Review by guotuofeng — APPROVED** [review]

File: olive/data/component/dataset.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        """
+        pass
+
+    def to_torch_tensor(self):
+        """
+        This function is used to convert the dataset to torch tensor
+        """
+        pass
+
+    def to_snap_dataset(self):
```
Comment (nbcsm) [inline olive/data/component/dataset.py:48]: typo, snpe

Comment (trajepl) [reply olive/data/component/dataset.py:48]: https://github.com/microsoft/Olive/pull/378 Updated in this PR.

**Pull request merged**